gohai.go

```diff
@@ -20,16 +22,24 @@ type Collector interface {
 	Collect() (interface{}, error)
 }
 
+type SelectedCollectors map[string]bool
```

Comment:
nitpick, a set should be implemented as `map[string]struct{}` instead. Avoids allocating booleans.